client/rest-high-level/src/main/java/org/elasticsearch/client/indexlifecycle/DeleteLifecycleRequest.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.indexlifecycle;
+
+import org.elasticsearch.client.MasterTimeoutRequest;
+
+import java.util.Objects;
+
+public class DeleteLifecycleRequest extends MasterTimeoutRequest<DeleteLifecycleRequest> {
```

Comment:
The transport based APIs extend `AcknowledgedRequest` which has a `timeout` and `ackTimeout` in addition to the `masterNodeTimeout`.

Curious if is this is intentional ?

The javadoc for `AcknowledgedRequest` states `Facilitates consistency across different api.` ... is that something we should carry forward while still divorcing ourselves from the transport request objects ?

Reply:
I based `MasterTimeoutRequest` on `MasterNodeRequest`, rather than `AcknowledgedRequest`, which doesn't have an `ackTimeout`.  It's possible I should be using `AcknowledgedRequest` instead - the difference wasn't immediately clear.

So, to answer your question: It was intentional, but it's possible I intended the wrong thing.

Reply:
I talked to @hub-cap on Slack and his answer was that these two timeouts should be based on the parameters that the requests can take.  To determine this, there should be a `Rest<whatever>Action` class which defines those parameters.  In this case, `RestDeleteLifecycleAction` (see [here](https://github.com/elastic/elasticsearch/blob/57835452228ba03238ef39e19829a67581689843/x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/indexlifecycle/action/RestDeleteLifecycleAction.java#L35)) defines both `timeout` and `master_timeout`, so this class should include both.

(Just waiting on some more feedback to make this change)

Reply:
@gwbrown Thanks for following up. I am working a similar issue and will follow your lead here.